frontend/src/plugins/core/componentFactory.tsx

```diff
@@ -241,6 +241,7 @@ export function registerReactComponent<T>(plugin: IPlugin<T, unknown>): void {
 
     disconnectedCallback() {
       this.observer.disconnect();
+      this.root?.unmount();
```

Comment:
this was a pretty big memory leak. i am going to pull this out into a different PR, with the debug tools